libs/partners/zenguard/langchain_zenguard/tools.py

```diff
@@ -0,0 +1,87 @@
+import os
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+import requests
+from langchain_core.pydantic_v1 import BaseModel, Field, ValidationError, validator
+from langchain_core.tools import BaseTool
+
+
+class Detector(str, Enum):
+    ALLOWED_TOPICS = "allowed_subjects"
+    BANNED_TOPICS = "banned_subjects"
+    PROMPT_INJECTION = "prompt_injection"
+    KEYWORDS = "keywords"
+    PII = "pii"
+    SECRETS = "secrets"
+    TOXICITY = "toxicity"
+
+class DetectorAPI(str, Enum):
+    ALLOWED_TOPICS = "v1/detect/topics/allowed"
+    BANNED_TOPICS = "v1/detect/topics/banned"
+    PROMPT_INJECTION = "v1/detect/prompt_injection"
+    KEYWORDS = "v1/detect/keywords"
+    PII = "v1/detect/pii"
+    SECRETS = "v1/detect/secrets"
+    TOXICITY = "v1/detect/toxicity"
+
+class ZenGuardInput(BaseModel):
+    prompts: List[str] = Field(..., min_items=1, min_length=1)
+    detectors: List[Detector] = Field(..., min_items=1)
+    in_parallel: bool = Field(default=True)
+
+class ZenGuardTool(BaseTool):
+    name = "ZenGuard"
+    description = "Fastest LLM Security Guardrails"
+    args_schema = ZenGuardInput
+    return_direct = True
+
+    zenguard_api_key: Optional[str] = Field(default=None)
```

Comment:
why this is optional?

Reply:
For the client’s convenience, he can pass the API key directly when creating a tool through a class, or he can set the API key to an environment variable and when creating an object from the class, it is taken from there
Example:
```
os.environ["ZENGUARD_API_KEY"] = <your_api_key>
tool = ZenGuardTool()
```
```
tool = ZenGuardTool(zenguard_api_key=<your_api_key>)
```